### PR TITLE
feature: Support google cloud builder.

### PR DESCRIPTION
**Pull request by wichopy**
Wanted to use the sentry-cli docker image in my company's Google Cloud Platform CI pipeline but ran into an issue trying to use the sentry-cli `set-commits --auto` command. The repository was not recognized similar to issue: https://github.com/getsentry/sentry/issues/11055

This PR parses the git information that google uses for its cloud builder.

### CODE REVIEW

**Comment by kamilogorek** [comment]
Closing, as it's from almost 2y ago. Will reopen if necessary.

**Comment by Datamance** [comment]
@kamilogorek Can we reopen this? If it's already supported, it'd be nice to have the preferred method explicitly outlined in the documentation.

**Comment by Datamance** [comment]
It's strange because I get `error: Could not automatically determine release name` during cloud builds, but when I use `cloud-build-local` to test locally the release is generated perfectly fine....

```Starting Step #7 - "create-sentry-release"
Step #7 - "create-sentry-release": Already have image (with digest): getsentry/sentry-cli
Step #7 - "create-sentry-release": Created release 5f7f963630de36da036624a026cee6934b39e837.
Step #7 - "create-sentry-release": +-------------------+--------------+
Step #7 - "create-sentry-release": | Repository        | Revision     |
Step #7 - "create-sentry-release": +-------------------+--------------+
Step #7 - "create-sentry-release": | lookit/lookit-api | 5f7f963630de |
Step #7 - "create-sentry-release": +-------------------+--------------+
Step #7 - "create-sentry-release": Finalized release 5f7f963630de36da036624a026cee6934b39e837.
Finished Step #7 - "create-sentry-release"```

File: src/utils/vcs.rs [hunk 1]
```diff
@@ -579,6 +594,12 @@ fn test_url_parsing() {
             id: "gitlab-org/gitlab-ce".into(),
         }
     );
+    assert_eq!(
+        VcsUrl::parse("https://source.developers.google.com/p/project-slug/r/github_org-slug_repo-slug"),
+        VcsUrl {
+            provider: "source.developers.google.com".into(),
+            id: "org-slug/repo-slug".into(),
+    )
```
Comment (kamilogorek) [inline src/utils/vcs.rs:602]:
```suggestion
    );
```

**Comment by kamilogorek** [comment]
@Datamance reopened, patched and merged. Will release later this week.

**Pull request merged**